crates/bevy_render/src/camera/projection.rs

```diff
@@ -134,54 +134,76 @@ impl Default for PerspectiveProjection {
+    /// The argument is the desired width of the projection in world units.
     FixedHorizontal(f32),
 }
 
+/// Project a 3D space onto a 2D surface using parallel lines, i.e., unlike [`PerspectiveProjection`],
+/// the size at which objects appear remain the same regardless of depth.
+///
+/// The volume contained in the projection is called the *view frustum*. Since the viewport is rectangular
+/// and projection lines are parallel, the view frustum takes the shape of a rectangular prism.
+///
+/// Note that the scale of the projection and the apparent size of objects are inversely proportional.
+/// As the size of the projection increases, the apparent size of objects decreases.
 #[derive(Component, Debug, Clone, Reflect, FromReflect)]
 #[reflect(Component, Default)]
 pub struct OrthographicProjection {
-    pub left: f32,
-    pub right: f32,
-    pub bottom: f32,
-    pub top: f32,
+    /// The distance of the near clipping plane in world units.
+    ///
+    /// Objects closer than this will not be rendered.
     pub near: f32,
+    /// The distance of the far clipping plane in world units.
+    ///
+    /// Objects further than this will not be rendered.
     pub far: f32,
-    pub window_origin: WindowOrigin,
+    /// Specifies the origin of the viewport as a normalized position from 0 to 1, where (0, 0) is the bottom left
+    /// and (1, 1) is the top right. This determines where the camera's position sits inside the viewport.
+    /// Set this to (0.5, 0.5) to make scaling affect opposite sides equally, thereby keeping centered objects on each axis centered.
+    ///
+    /// When the projection scales due to viewport resizing (assuming `scaling_mode` is not set to `Fixed`), the position
+    /// of the camera doesn't change, and since `viewport_origin` specifies the point on the viewport where the camera sits,
+    /// this point will always remain at the same position (relative on the viewport).
+    ///
+    /// Consequently, this is pivot point when scaling. With a bottom left pivot, the projection will expand
+    /// upwards and to the right. With a top right pivot, the projection will expand downwards and to the left.
+    /// Values in between will caused the projection to scale proportionally on each axis.
+    pub viewport_origin: (f32, f32),
+    /// How the projection will scale when the viewport is resized.
     pub scaling_mode: ScalingMode,
+    /// Scales the projection, in world units.
+    ///
+    /// As scale increases, the apparent size of objects decreases, and vice versa.
     pub scale: f32,
+    left: f32,
+    right: f32,
+    bottom: f32,
+    top: f32,
 }
 
 impl CameraProjection for OrthographicProjection {
     fn get_projection_matrix(&self) -> Mat4 {
         Mat4::orthographic_rh(
-            self.left * self.scale,
-            self.right * self.scale,
-            self.bottom * self.scale,
-            self.top * self.scale,
+            self.left,
```

Comment:
Behavior change?

Reply:
I think it makes more sense now since `left`, `right`, etc. can't be modified directly anymore. It would be less consistent if every other field affected these values but `scale` did not.